skbio/alignment/_tabular_msa.py

```diff
@@ -1847,59 +1892,89 @@ def extend(self, sequences, minter=None, index=None):
         ACGT
         AG-T
         -G-T
+        >>> msa.index
+        Index(['seq1', 'seq2', 'seq3'], dtype='object')
 
-        Auto-incrementing index labels:
+        Extend with more sequences, this time resetting the MSA's index labels
+        to the default with `reset_index`. Note that since the MSA's index is
+        reset, we do not need to provide index labels for the new sequences via
+        `index` or `minter`:
 
+        >>> msa.extend([DNA('ACGA'), DNA('AC-T'), DNA('----')],
+        ...            reset_index=True)
+        >>> msa
+        TabularMSA[DNA]
+        ---------------------
+        Stats:
+            sequence count: 6
+            position count: 4
+        ---------------------
+        ACGT
+        AG-T
+        ...
+        AC-T
+        ----
         >>> msa.index
-        Int64Index([0, 1, 2], dtype='int64')
-        >>> msa.extend([DNA('ACGA'), DNA('AC-T'), DNA('----')])
-        >>> msa.index
-        Int64Index([0, 1, 2, 3, 4, 5], dtype='int64')
+        RangeIndex(start=0, stop=6, step=1)
 
         """
-        if minter is not None and index is not None:
+        if sum([minter is not None, index is not None, reset_index]) != 1:
```

Comment:
use `bool(reset_index)` as `sum` will act weird in a truthy context. Assuming with allow `reset_index` to ducktype its boolean-ness